smk/include/bwa_index_wrapper.smk

```diff
@@ -0,0 +1,226 @@
+# a 'status' file that denotes sync has been done on a different node. We have a simple solution for this.
+# The node names in CLUSTER_NODES should match the node name accessible by the command 'hostname --short'.
+# The sync status files are named with the local node name. You ask for sync'ing in nodeX, and
+# only nodeX can make that file. This avoids problems when this is NOT run using --cluster.
+# There is anothe problem that multiple sync rules may run the same 'rsync source dest' command
+# on a single server. This would likely cause a race condition. To avoid that, we create a lock
+# using 'flock' so that only one rsync is running for a specific bwa index sync on one node.
+# As a bonus, this also allows rsyncing an NFS bwa index onto a local disk when running without
+# --cluster, if you define CLUSTER_NODES with just your exec node's name.
+################################################################################################
+
+if CLUSTER_WORKLOAD_MANAGER is not None:
+    known_managers = ['SLURM', 'PBS', 'TORQUE', 'SGE', 'LSF']
+    if CLUSTER_WORKLOAD_MANAGER.upper() not in known_managers:
+        raise Exception(f"MIntO error: Unexpected value: CLUSTER_WORKLOAD_MANAGER='{CLUSTER_WORKLOAD_MANAGER}'. Must be one of {known_managers}")
+
+if CLUSTER_NODES is not None:
+
+    # Make sure other variables are defined
+    if CLUSTER_WORKLOAD_MANAGER is None or CLUSTER_LOCAL_DIR is None:
+        raise Exception(f"MIntO error: CLUSTER_WORKLOAD_MANAGER and CLUSTER_LOCAL_DIR must be defined, if CLUSTER_NODES is defined")
+
+    # Baseline    : 5 GB
+    # Size-based  : 22 byte per byte file size
+    # New attempts: +40 GB each time
+    rule BWA_index_contigs_nfs:
+        input:
+            fasta="{somewhere}/{something}.{fasta}"
+        output:
+            "{somewhere}/BWA_index/{something}.{fasta}.0123",
+            "{somewhere}/BWA_index/{something}.{fasta}.amb",
+            "{somewhere}/BWA_index/{something}.{fasta}.ann",
+            "{somewhere}/BWA_index/{something}.{fasta}.bwt.2bit.64",
+            "{somewhere}/BWA_index/{something}.{fasta}.pac",
+        log:
+            "{somewhere}/BWA_index/BWA_index.{something}.{fasta}.log"
+        wildcard_constraints:
+            fasta='fasta|fna'
+        resources:
+            mem = lambda wildcards, input, attempt: 5 + int(22*input.size_mb/1024) + 40*(attempt-1),
+        threads: 4
+        conda:
+            config["minto_dir"]+"/envs/MIntO_base.yml" #bwa-mem2
+        shell:
+            """
+            outdir=$(dirname {output[0]})
+            umask 002 && mkdir -p {CLUSTER_LOCAL_DIR}/$outdir
+            cd {CLUSTER_LOCAL_DIR}/$outdir
+            prefix=$(basename {input})
+            time (
+                bwa-mem2 index {input.fasta} -p $prefix
+                echo "NODE: $(hostname)"
+                echo "SENDING:"
+                echo "$(pwd)/$prefix --> $outdir"
```

Comment:
` index` command resolves the index file path from `-p` relative to `pwd`. Default value for `-p` is the input fasta file itself. This is probably what you experienced: when you don't specify `-p`, then index is created in the input folder because that's what `-p` resolves to. Since we run it after doing `cd {CLUSTER_LOCAL_DIR}/$outdir` and use `$(basename {input})` as value for `-p`, it just uses the filename of input but places the index in `pwd`.